File: Java/NetBeans/SV/src/Exe_DSSV/SVTH.java
package Exe_DSSV;

public class SVTH extends SV{

    protected String tenDT;
    protected double diemDT;

    public SVTH(String hoTen, int namSinh, double dtb, String tenDT, double diemDT) {
        super(hoTen, namSinh, dtb);
        this.tenDT = tenDT;
        this.diemDT = diemDT;
    }
    
    public void hienThi() {
        super.hienThi();
        System.out.println("Ten dao tao: " + tenDT);
        System.out.println("Diem dao tao: " + diemDT);
    }
    
    public String loaiSV() {
        return "TH";
    }
    
    public boolean equals(Object sv) {
        if (sv != null && sv instanceof SVTH b) {
            return super.equals(b)
                    && this.tenDT.equals(b.tenDT)
                    && diemDT == b.diemDT;
        }
        return false;
    }
    
    public String toString() {
        return super.toString() + "\n" +
               "Ten dao tao: " + tenDT + "\n" +
               "Diem dao tao: " + diemDT;
    }
    
    public boolean duocTN() {
        return dtb >= 5.0f && diemDT >= 7.0f;
    }
    
    
}

File: C#/QLSVTHvaSP/QLSVTHvaSP/SVTH.cs
namespace QLSVTHvaSP
{
    internal class SVTH : SV
    {
        private string _tenDT;
        private float _diemDT;

        public SVTH() : base() { }

        public SVTH(string hoTen, int namSinh, float diemTB, string tenDT, float diemDT) : base(hoTen, namSinh, diemTB)
        {
            _tenDT = tenDT;
            _diemDT = diemDT;
        }

        public string TenDT { get => _tenDT; set => _tenDT = value; }
        public float DiemDT { get => _diemDT; set => _diemDT = value; }

        public override void HienThi()
        {
            base.HienThi();
            Console.Write($", {_tenDT}, {_diemDT}");
        }

        public override bool DuocTN()
        {
            return DiemTB >= 5 && _diemDT >= 5;
        }

        public override string LoaiSV()
        {
            return "TH";
        }
    }
}